patches/server/0854-Added-and-implemented-PlayerLootDropEvent.patch

```diff
@@ -0,0 +1,52 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: NeonOverflow <nneonoverflow@protonmail.com>
+Date: Thu, 13 Jan 2022 22:58:28 -0500
+Subject: [PATCH] Added and implemented PlayerLootDropEvent.
+
+As a result of how the PlayerDeathEvent was implemented, when a player dies their items are simply dropped
+through the "world.dropItem()" method. This makes it very annoying for plugin developers to ultimately
+track down the Item entities dropped when a player dies. This patch implements a "PlayerLootDropEvent",
+making it easy for developers to track down the Item entities dropped when a player dies. Furthermore,
+Item entities dropped on player death are marked as being thrown by the player that died.
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+index 35cc150adf51f79e2fccef8b094c90554aafbee2..c484948faf336ca0efb4ce4201c41b749e355483 100644
+--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
++++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+@@ -907,15 +907,35 @@ public class CraftEventFactory {
+         victim.expToDrop = event.getDroppedExp();
+         victim.newExp = event.getNewExp();
+ 
++        // Paper start - implement player loot drop event + set item thrower as dead player
++        List<Item> items = new ArrayList<Item>();
++
+         for (org.bukkit.inventory.ItemStack stack : event.getDrops()) {
+             if (stack == null || stack.getType() == Material.AIR) continue;
+ 
+-            world.dropItem(entity.getLocation(), stack);
++            Item item = world.dropItem(entity.getLocation(), stack);
++            item.setThrower(victim.getUUID());
```

Comment:
This is a behavioral change. In vanilla, item's dropped by players do not have their thrower set. They specifically don't. The thrower, in vanilla, is used for the `thrown_item_picked_up_by_entity` advancement trigger, and items dropped on death are not counted as "thrown by the player dying", only items intentionally dropped.